### DISCUSSION TITLE
Buffer list not displayed when there is broken tramp buffer amongst them

### DISCUSSION THREAD

**Discussion by jumper047** [minad/marginalia#188] [Q&A]
So, I have an dired buffer displaying some directory on remote box. Then I'm losing connection to that box. And after that, if I'm trying to switch buffer with marginalia mode enabled, I got this error in echo area after some waiting: 
```
Tramp: Opening connection nil for username@192.168.1.233 using scp...failed
Error in post-command-hook (vertico--exhibit): (file-error "Tramp failed to connect.  If this happens repeatedly, try
    ‘M-x tramp-cleanup-this-connection’")
```
There are no buffer list displayed in vertico until marginalia mode is enabled.

**Comment by minad** [comment]
Which part of Marginalia breaks? See https://github.com/minad/vertico?tab=readme-ov-file#debugging-vertico for obtaining a stack trace. I am happy to accept a patch which hardens Marginalia accordingly. Besides that I recommend to filter recentf to not include remote files.

EDIT: I just tried to reproduce the problem, but it doesn't occur in my setup.

**Comment by ferrine** [comment]
I observe exactly the same issue. I have buffers that refer to the remote server (connection lost) and it causes marginalia to fail. Interestingly it happens once the selection candidate is the remote one.

**Reply by minad** [reply]
Can you please provide a stack trace?

**Reply by ferrine** [reply]
unfortunately, it is quite challenging to reproduce it

**Comment by ferrine** [comment]
it happens under some uncertain conditions, can't reproduce it. Could it be a solution (implemented in the library) to output errors in some Warnings or Messages buffer instead of blocking the entire minibuffer interaction?

**Reply by ferrine** [reply]
I had it under my lost session and so far my fix it is to wrap vertico--exhibit with ignore errors advice

**Reply by minad** [reply]
See https://github.com/minad/vertico?tab=readme-ov-file#debugging-vertico

**Comment by pedro-nonfree** [comment]
I can consistently reproduce the bug (I was trying to do a `tramp-cleanup-all-connections` when I am putting my laptop on suspend/etc because it stops wifi, and might cause emacs hanging when it recovers because of previously opened remote connections such as with tramp ssh )

```lisp
(setq my/ssh-path "file:/ssh:my-ssh-host:~/")
(setq my/wifi "\"My Wifi ESSID\"")

(org-link-open-from-string my/ssh-path)
;; disable wifi using nmcli (you can stop wifi/ethernet/internet whatever way you use it)
(shell-command-to-string (concat "nmcli connection down " my/wifi))
(tramp-cleanup-all-connections)
(consult-buffer)
```

how to generate the stack trace?

seeing I have no clue how to do it, can you say me exact emacs lisp script to add on top of the previous?

This is my `*Messages*` buffer (I tweaked a little bit saying on what buffer originated the error and the timestamp - 1activity is my journal file, where I am executing that emacs lisp script; board is like the `*scratch*`/initial buffer)

```
"Connection 'MY WIFI' successfully deactivated (D-Bus active path: /org/freedesktop/NetworkManager/ActiveConnection/40)
"
[2026-05-25T01:44:22.258] [ 1activity.org ] 
Cleaning up the recentf list...
[2026-05-25T01:44:22.261] [ 1activity.org ] Cleaning up the recentf list...done (0 removed)
nil
[2026-05-25T01:44:23.279] [ 1activity.org ] 
Tramp: Opening connection for my-ssh-host using ssh...
[2026-05-25T01:44:23.291] [ 1activity.org ] Tramp: Opening connection for my-ssh-host using ssh...failed
[2026-05-25T01:44:23.292] [ 1activity.org ] Cleaning up the recentf list...
[2026-05-25T01:44:23.294] [ 1activity.org ] Cleaning up the recentf list...done (0 removed)
[2026-05-25T01:44:23.295] [ 1activity.org ] File error: Tramp failed to connect.  If this happens repeatedly, try
    ‘M-x tramp-cleanup-this-connection’
Error in post-command-hook (vertico--exhibit): (file-error "Tramp failed to connect.  If this happens repeatedly, try
    ‘M-x tramp-cleanup-this-connection’")
[2026-05-25T01:44:25.188] [ board.org ] Tramp: Opening connection for my-ssh-host using ssh...
[2026-05-25T01:44:25.200] [ board.org ] Tramp: Opening connection for my-ssh-host using ssh...failed
[2026-05-25T01:44:25.201] [ board.org ] Cleaning up the recentf list...
[2026-05-25T01:44:25.204] [ board.org ] Cleaning up the recentf list...done (0 removed)
[2026-05-25T01:44:25.205] [ board.org ] File error: Tramp failed to connect.  If this happens repeatedly, try
    ‘M-x tramp-cleanup-this-connection’
Error in post-command-hook (vertico--exhibit): (file-error "Tramp failed to connect.  If this happens repeatedly, try
    ‘M-x tramp-cleanup-this-connection’")
```
